docs/source/advanced.rst

```diff
@@ -0,0 +1,354 @@
+information only, they do not provide any additional functionality on top of
+what the :samp:`SQLAlchemy` model provides. They are primarily
+used to enable IDE auto-complete and type hint support. The models can be used
+in the same way as the models that can be imported from
+:samp:`open_alchemy.models` and provide the full functionality of
+:samp:`SQLAlchemy` models.
+
+The following is a sample file generated for the above example:
+
+.. literalinclude:: ../../examples/simple/models_auto.py
+    :language: python
+    :linenos:
+
+The following information is recorded in the models file:
+
+* The name and type of each property of a schema.
+* The :ref:`from-dict` and :ref:`to-dict` function signatures, including
+  the type of the arguments and return values.
+* The :ref:`from-str` and :ref:`to-str` function signatures, including
+  the type of the arguments and return values.
+* The properties created on instance objects due to any :ref:`backref`.
+* Special SQLAlchemy properties for interacting with the database.
+* The object and property descriptions from the OpenAPI specification in the
+  class and function docstrings.
+
+.. _backrefs:
+
+.. note:: To be able to add relationships created by :samp:`x-backrefs` to the
+    type annotations of the models file, the schema stored alongside a model,
+    which is accessible at the :samp:`_schema` class variable (not a public
+    interface so it should not be used or relied upon), will use the
+    :samp:`x-backrefs` extension property to record the schema for all back
+    references for the model. :samp:`x-backrefs` is not a public interface and
+    should not be relied upon as it is subject to change.
+
+It is worth noting that the auto-generated models cannot be imported directly.
+The generated models are added to the :samp:`open_alchemy.models` package by the
+:samp:`init_*` functions.
+
+.. code-block:: python
+  :linenos:
+
+  from open_alchemy import init_yaml
+  from open_alchemy import models
+
+  init_yaml('openapi.yml')
+  employee = models.Employee(name="David Andersson", division="Engineering")
+
+To use the models in another project, refer to the :ref:`build-yaml` or
+:ref:`build-json` functions.
+
+.. _model-utilities:
+
+Model Utilities
+---------------
+
+There are a few common utility functions that are added to the models. The
+:ref:`from-dict` utility function constructs a model instance from a
+dictionary. The :ref:`to-dict` function converts a model instance to a
+dictionary.
+
+.. _from-dict:
+
+:samp:`from_dict`
+^^^^^^^^^^^^^^^^^
+
+The :samp:`from_dict` function is available on all constructed models. It
+accepts a dictionary and constructs a model instance based on the dictionary.
+It is similar to :python:`Employee(**employee_dict)` with a few advantages:
+
+* The dictionary based on which the model is constructed is checked against
+  the schema used to define the model.
+* If the model includes a relationship, the relationship is constructed
```

Comment:
btw, this is another advantage of `from_dict`, the case where destructing a dictionary into a model constructor does not work is in the case of a relationship, where the relationship parameter needs to be the model and not the dictionary representation of it, but it is possible to destruct a dictionary into `from_dict` in that case as well